pyproject.toml

```diff
@@ -118,6 +118,7 @@ rich = "^13.7.1"
 psycopg2 = { version = "^2.9.9", optional = true }
 pgvector = { version = "^0.2.5", optional = true }
 structlog = "^24.1.0"
+llama-index = "^0.10.30"
```

Comment:
don't you think it should be optional?

Reply:
I know it is causing lots of confusion but for the dependencies right now we use 3 different places... I have a PR to fix that in a short but if it is going to be merge before that you need to add your dependencies in requirements.txt (which use for building our package), here that you added it, and `dependencies` in the pyproject.toml ...

 more info:
https://github.com/stanfordnlp/dspy/pull/819

Reply:
I'm open to making the dependency optional - but wouldn't my tests fail as you pointed out if we run that route?

Reply:
I think the best way is to tag those tests to just be ran when the llama-index is installed and we fix the CI based on that.

Reply:
I'll go with that. I'll make the adjustment today or tomorrow. I'll only run those tests when llamaindex is present.